### PR TITLE
WIP: AdobeSign integration

### PR DESCRIPTION
**Pull request by Emily-ejag**
## Proposed changes

Please first link this branch with this PR on firekit: [AdobeSign Firekit](https://github.com/yeatmanlab/roar-firekit/pull/128)

-- Adding identical files from AdobeSign to the Legal section.
-- Introducing an Adobe Sign checkbox in the Consent Picker, saving the legal documents under the legal directory in the administration.
-- On the homeParticipant check, verify if the adobeSign flag exists. If it does, it ask to input your email, then you have 20 retrys of 5 sec each to sign, if not you need to start the process again.
-- If the signing is successful then the student document gets updated with the consent form

ON PARENT SIGNUP
for using adobe on parent signup the route is roar.education/register-clinic

## Types of changes

What types of changes does this pull request introduce?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (non-breaking change that does not add functionality but makes code cleaner or more efficient)
- [ ] Documentation Update
- [ ] Tests (new or updated tests)
- [ ] Style (changes to code styling)
- [ ] CI (continuous integration changes)
- [ ] Repository Maintenance
- [ ] Other (please describe below)

## Checklist

- [x] I have read the [guidelines for contributing](https://github.com/yeatmanlab/roar-dashboard/blob/main/.github/CONTRIBUTING.md).
- [x] The changes in this PR are as small as they can be. They represent one and only one fix or enhancement.
- [x] Linting checks pass with my changes.
- [x] Any existing unit tests pass with my changes.
- [x] Any existing end-to-end tests pass with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] If this PR fixes an existing issue, I have added a unit or end-to-end test that will detect if this issue reoccurs.
- [ ] I have added JSDoc comments as appropriate.
- [ ] I have added the necessary documentation to the [roar-docs repository](https://github.com/yeatmanlab/roar-docs).
- [ ] I have shared this PR on the roar-pr-reviews channel (if I have access)
- [ ] I have linked relevant issues (if any)

### CODE REVIEW

File: src/helpers/query/adobeSign.js
Comment (richford) [inline src/helpers/query/adobeSign.js:1]: We discussed in person how this functionality should be moved into cloud functions and then called via firekit functions.

File: src/pages/HomeParticipant.vue [hunk 1]
```diff
@@ -86,6 +87,47 @@
...
+        </div>
+        <div v-else class="bg-white p-3 border-1 border-round w-full">
+          <h2>Please review your email to sign the document and proceed</h2>
+          <div class="justify-content-center flex">
+            <img src="../assets/signDoc2.webp" style="width: 400px; border-radius: 0.5rem" />
+          </div>
+        </div>
+      </template>
+    </PvDialog>
+  </div>
```
Comment (richford) [inline src/pages/HomeParticipant.vue:130]: Can you move this to a separate `AdobeSignDialog.vue` component?

**Review by richford — CHANGES_REQUESTED** [review]

File: src/components/ConsentPicker.vue [hunk 2]
```diff
@@ -18,6 +18,16 @@
         >This Administration does not require a consent/assent</label
       >
     </div>
+    <div class="flex justify-content-center mt-2">
+      <PvCheckbox
+        v-model="adobeSign"
+        input-id="no-consent"
+        class="flex"
+        value="adobeSign"
```
[the remark this answers is not in the document]
Comment (Emily-ejag) [reply src/components/ConsentPicker.vue:26]: Yes, adobeSign is just for show or not show the dialog, and isAdobeSign is for actually saving the value in the database if it was successful. I think the confusion will go away once I do a separate component for AdobeSign

File: src/pages/HomeParticipant.vue (hunk 1 above)
Comment (Emily-ejag) [reply src/pages/HomeParticipant.vue:130]: Yes!

File: src/helpers/query/adobeSign.js
Comment (Emily-ejag) [reply src/helpers/query/adobeSign.js:1]: Yes! Doing that

**Review by richford — APPROVED** [review]

**Comment by richford** [comment]
@Emily-ejag , we've moved the integration token to Google Secret Manager. But there are still the old integration tokens in the git commit history. Can you look through the [tokens identified here](https://github.com/yeatmanlab/roar-dashboard/pull/704/checks?check_run_id=27394595116) and delete them from your adobe sign developer account. Then we can dismiss the GitGuardian checks that are failing.

**Comment by Emily-ejag** [comment]
Created a new PR (https://github.com/yeatmanlab/roar-dashboard/pull/716) to not have security issues

**Pull request closed without merging**